### DISCUSSION TITLE
discord.ext.commands - get attrs Arguments of Command

### DISCUSSION THREAD

**Discussion by niklasmoell** [Rapptz/discord.py#7433] [Q&A] [answered]
Hello,
how can I get a attrs Keyword Argument from a Command?

For example:

`@commands.command(name="test", usage="my usage") `

How can I get the `usage` attribute now?

**Comment by HarryPotter12352** [comment]
The way you would use an attr, I should think.
`name.usage`
That's how prettyhelp works too.

**Comment by AkshuAgarwal** [comment, marked as answer]
You can get the Command using `bot.get_command('your_command_name')` and use `cmd.usage` (where `cmd` is returned by `get_command`).

Alternatively, you can also use `.usage` on the decorated function. Like:
```py
>>> @commands.command(usage='<usage>')
>>> async def foo(self, ctx): ...
>>>
>>> foo.usage
'<usage>'
```

Documentation: https://discordpy.readthedocs.io/en/stable/ext/commands/api.html#discord.ext.commands.Command.usage

**Reply by niklasmoell** [reply]
Thanks a lot!

**Reply by imvaskel** [reply]
To add onto this, using the `usage` kwarg will override the `signature` attribute, which by default is a posix-style signature of the command.
